main.go

```diff
@@ -148,6 +152,11 @@ func main() {
 		oauth2Config:            oauth2Config,
 	}
 
+	gcpAuthenticator, err := newGcpAuthenticator(c.GCPHeader)
+	if err != nil {
+		log.Fatalf("error initializing gcp authenticator")
```

Comment:
It would be useful to also emit the actual error
```
log.Fatalf("error initializing gcp authenticator: %v", err)
```

Reply:
Actually, do we want it to totally fail if it cannot load the gcp authenticator? Or just log and move on...

```
time="2021-08-31T09:05:06Z" level=fatal msg="error initializing gcp authenticator: error creating validator: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information."
```

Reply:
Actually failing on load isn't great either, since then you will have problems in server.go:103 at runtime...

Reply:
maybe we can have something like this in main.go
```
	authenticators := []Authenticator{sessionAuthenticator, idTokenAuthenticator}
	if gcpAuthenticator != nil {
		authenticators = append(authenticators, gcpAuthenticator)
	}

...

		caBundle:                caBundle,
		authenticators:          authenticators,
		authorizers:             []Authorizer{groupsAuthorizer},
```